Assets/Common/Scripts/DataTable/TableBase.cs

```diff
@@ -0,0 +1,22 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+
+
+public struct ObjectData
+{
+    // "Normal_((0, 0), (3, 0))"
+    public string Type;
+    public int Size;
+    public List<Tuple<int, int>> Pos;
+}
+
+
+public abstract class TableBase
+{
+    public abstract void SetTable(string key, string name, string value);
+
+    protected Dictionary<string, object> mDicData = new Dictionary<string, object>();
+    public Dictionary<string, object> DicData => this.mDicData;
+}
```

Comment:
딕셔더리의 값이 object이면 boxing, unboxing 이슈가 있을 듯 합니다.
클래스를 generic으로 바꾸고 그 타입을 값으로 넣어주는 것도 방법일 것 같네요.
```suggestion
public abstract class TableBase<T>
{
    public abstract void SetTable(string key, string name, string value);

    protected Dictionary<string, T> mDicData = new Dictionary<string, T>();
    public Dictionary<string, T> DicData => this.mDicData;
}
```
이런 식인데 StageTable의 객체를 만든다면 T에 StageData 타입을 넣어서 생성하면 될 것 같아요.

Reply:
제네릭 타입이 되고 리플렉션으로 필드 값 자동으로 주입이 된다면 자식 클래스가 필요 없어질 수도 있겠네요.
근데 이건 확실친 않습니다. ^_^;